system/scripts/zone/core/calc_combat.cs

```diff
@@ -73,19 +75,42 @@ public float SCR_CalculateDamage(ICombatEntity attacker, ICombatEntity target, S
 
 		var rnd = RandomProvider.Get();
 
-		var dodgeChance = SCR_GetDodgeChance(attacker, target, skill, skillHitResult);
+		var dodgeChance = SCR_GetDodgeChance(attacker, target, skill, modifier, skillHitResult);
 		if (rnd.Next(100) < dodgeChance)
 		{
 			skillHitResult.Result = HitResultType.Dodge;
 			return 0;
 		}
 
-		var damage = SCR_GetRandomAtk(attacker, target, skill, skillHitResult);
+		var damage = SCR_GetRandomAtk(attacker, target, skill, modifier, skillHitResult);
+
+		// Check for Concentrate
+		if (attacker.Components.Get<BuffComponent>().TryGet(BuffId.Concentrate_Buff, out var concentrateBuff))
+		{
+			var bonusDamage = concentrateBuff.NumArg2;
+			var variableName = "Melia.HitsLeft";
+			if (concentrateBuff.Vars.TryGetFloat(variableName, out var hitsLeft))
+			{
+				hitsLeft--;
+				if (hitsLeft > 0)
+					concentrateBuff.Vars.SetFloat(variableName, hitsLeft);
+				else
+					attacker.Components.Get<BuffComponent>().Remove(BuffId.Concentrate_Buff);
+			}
+			modifier.BonusDamage += bonusDamage;
+		}
+
+		// Check for Dagger Slash
+		if (attacker.Components.Get<BuffComponent>().TryGet(BuffId.DaggerSlash_Buff, out var daggerSlashBuff))
+		{
+			modifier.DamageMultiplier += daggerSlashBuff.OverbuffCounter * 0.07f;
+		}
 
 		var skillFactor = skill.Properties.GetFloat(PropertyName.SkillFactor);
 		var skillAtkAdd = skill.Properties.GetFloat(PropertyName.SkillAtkAdd);
 		damage *= skillFactor / 100f;
-		damage += skillAtkAdd;
+		damage += skillAtkAdd + modifier.BonusDamage;
```

Comment:
Note: Kinda wish the skill modifier had been its own PR, as that would've made it easier to review.

Reply:
I can see the argument for it, though then there wouldn't be any examples of its usage.

Reply:
You don't really need an example for this, it's all in the code. The example is just setting a variable and passing the modifier^^